api/models/constants.py

```diff
@@ -1,2 +1,69 @@
 DEFAULT_CHAR_FIELD_MAX_LEN = 255
 DEFAULT_CHOICE_CHAR_FIELD_MAX_LENGTH = 15
+
+# IMPORTANT: the boundaries of larger ranges must match the boundaries of the smaller ones
+
+# Match string ranges that are presented to the user to actual ranges.
+# A dict comprehension would be more concise but less readable.  Dict comprehension would also be
+# less error-prone, but the dict isn't supposed to be changed frequently (or at all) after launch.
+STUDENT_AGE_RANGES = {
+    "5-6": (5, 7),
```

Comment:
I don't understand why here and below you don't use (5, 6), (7, 8), ... etc.? Why key is 5-6, but value is 5-7? Could you explain this please? :)

Reply:
e.g. user chooses "5-6" and the bot passes this "5-6" to the back-end. But it actually means `range (5, 7)` because 5 years old is OK, 6 years old is OK, but 7 years old is too old. `range (5, 6)` would exclude `6`.

Of course, maybe I'm thinking too much. Maybe a DB query would be just "more or equal than 5 and less or equal than 6".

But then I should go back to the list comprehension.

Reply:
I don't think that it will be used in bare `range()`, but in orm-queries or direct comparing like `a < x < b`, so probably better to make them same. 

Also I still don't sure that you need generator here, because ranges are not same. For example, if you have identical range of 5 years, better to generate it like
`ranges = {f"{left}-{right}": (left, right) for left, right in zip(range(5, 15), range(6, 16))}`
(it will produce `{'5-6': (5, 6), '6-7': (6, 7), '7-8': (7, 8), ...}` to 15-16)

But if ranges are different, like here (from 5 to 6, but from 36 to 40), you will need to store plain ranges in code (as str-s or as tuples), and generate dict from them, so it will be more readable if you will just make dict (how it's made now)

Another option is to generate different parts of ranges (with different step), and then concatenate them.
But I still don't think that it's good idea :)